packages/core/lib/command-utils.js

```diff
@@ -107,7 +112,9 @@ const prepareOptions = ({ command, inputStrings, options }) => {
   });
 
   // method `extractFlags(args)` : Extracts the `--option` flags from arguments
-  const inputOptions = extractFlags(inputStrings);
+  const inputOptions = extractFlags(inputStrings).filter(
```

Comment:
We don't want to do this filter for all commands. It is reasonable that we would have a `--version` or `-v` flag for a command and in that case we wouldn't want to remove it here. We only would want to remove it in the case that they are trying to get the version info.

Hmm, it also seems like maybe `extractFlags` is not correct as it won't extract flags that begin with a single `-` such as `-v`.

Reply:
Can you help to share some examples of when the command would have `--version` or `-v`? Is it to get the version info for a certain truffle command?
When that happens, we will then remove the filter logic, no? And below it will check it as valid options. 

`extractFlags` only works with `--` right now. Should I add this change to this PR? or separate PR?

Reply:
We don't have a `--version` flag for any other commands. That being said, this is not the correct behavior. If we happen to add a `--version` flag to some other command in the future, we don't want to have to remember, "hey, there was that thing we added a bit ago that will now cause incorrect behavior and now we have to remember to remove it". That wouldn't be great. Instead we can just build in the correct behavior now.

We are making an exception for the `version` command here. We want this code to do exactly what we want and we don't want to create code that "just happens to work" for doing the thing we want in the short term. Make sense?

Reply:
As for the change to `extractFlags`, I would say make it a separate PR. It is easier when PRs are smaller and more focussed subject-wise :)